File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.cpp
class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int res = 0;
        vector<string> cols(strs[0].size(),"");
        for(auto s:strs)
            for(int i=0; i<cols.size();i++)
                cols[i]+=s[i];
        for(auto c:cols){
            string x = c;
            sort(x.begin(), x.end());
            if(x!=c)
                res++;
        }
        return res;
    }
};

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.java
class Solution {
    public int minDeletionSize(String[] strs) {
        int n = strs[0].length(), del = 0;
        loop: for (int i = 0; i < n; i++) {
            char prev = 'a';
            for (String s : strs) {
                char curr = s.charAt(i);
                if (curr < prev) {
                    del++;
                    continue loop;
                }
                prev = curr;
            }
        }
        return del;
    }
}